staging/vhost-device-console/src/console.rs

```diff
@@ -0,0 +1,63 @@
+// Console backend device
+//
+// Copyright 2023-2024 VIRTUAL OPEN SYSTEMS SAS. All Rights Reserved.
+//          Timos Ampelikiotis <t.ampelikiotis@virtualopensystems.com>
+//
+// SPDX-License-Identifier: Apache-2.0 or BSD-3-Clause
+
+use crate::vhu_console::VirtioConsoleConfig;
+use log::trace;
+use thiserror::Error as ThisError;
+
+type Result<T> = std::result::Result<T, Error>;
+
+#[derive(Copy, Clone, Debug, PartialEq, ThisError)]
+pub(crate) enum Error {}
+
+#[derive(Debug)]
+pub(crate) struct ConsoleController {
+    config: VirtioConsoleConfig,
+    pub exit: bool,
+}
+
+impl ConsoleController {
+    // Creates a new controller corresponding to `device`.
+    pub(crate) fn new() -> Result<ConsoleController> {
+        Ok(ConsoleController {
+            config: VirtioConsoleConfig {
+                cols: 20.into(),
+                rows: 20.into(),
+                max_nr_ports: 1.into(),
+                emerg_wr: 64.into(),
+            },
+            exit: false,
+        })
+    }
+
+    pub(crate) fn config(&self) -> &VirtioConsoleConfig {
+        trace!("Get config\n");
+        &self.config
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_console_controller_creation() {
+        let controller = ConsoleController::new();
+        assert!(controller.is_ok());
+    }
+
```

Comment:
Generally results should be `unwrap`ped or `expect`ed in tests, so that their error message shows up in the test results. Here though you already unwrap (the infallible as mentioned above) `ConsoleController::new()` in the `test_console_controller_config` so `test_console_controller_creation` is unnecessary.